src/main/java/org/jabref/logic/search/SearchQuery.java

```diff
@@ -85,7 +85,7 @@ private String getLocalizedRegularExpressionDescription() {
     }
 
     public boolean isGrammarBasedSearch() {
-        return rule instanceof GrammarBasedSearchRule;
+        return rule instanceof GrammarBasedSearchRule || rule instanceof RegexBasedSearchRule;
```

Comment:
Should regex really be treated as grammar-based search? I would not treat it like that, because regex search is orthogonal to grammar-based search. One can activate it in addition.

Reply:
At least in the documentation: https://help.jabref.org/en/Search#search-modes they are listed under the heading of "advanced search". But I can easily change that of course.

Reply:
I updated the help accordingly to also reflect that regular expressions are orthogonal. 😇 

Could you additionally add a comment there that this is the "Advanced Search" described in the help? I know, this contradicts [DDD](https://en.wikipedia.org/wiki/Domain-driven_design), but I think, it is OK in this place.

Reply:
Ok, I adapted the code accordingly.